packages/web-components/src/label/README.md

```diff
@@ -0,0 +1,131 @@
+# Label
+
+> A label represents a caption for an item in a user interface.
+
+<br />
+
+## **Design Spec**
+
+[Link to Label Design Spec in Figma](https://www.figma.com/file/aFHbqxQlX8fetFhyYl6STb/TF---Trident-control-specs?node-id=339%3A52012)
+
+<br />
+
+## **Engineering Spec**
+
+fluent-label is a web component implementation of an HTML label element. The fluent-label several visual font sizes (small, medium, large) and font weights (regular, semibold). The fluent-label also provides appearances for required and disabled states.
+
+### Use Case
+
+Creating a simple label element with an optional info icon and optional required state
+
+<br />
+
+## Class: `Label`
+
+<br />
+
+### **Variables**
+
+<br />
+
+### **Fields**
+
+| Name       | Privacy | Type                           | Default     | Description                                                       |
+| ---------- | ------- | ------------------------------ | ----------- | ----------------------------------------------------------------- |
+| `id`       | public  | `string`                       |             | Specifies the id of label                                         |
+| `for`      | public  | `string`                       |             | Specifies the id of the form element the label should be bound to |
+| `form`     | public  | `string`                       |             | Specifies which form the label belongs to                         |
+| `required` | public  | `boolean`                      | `false`     | Specifies required styling for label                              |
+| `disabled` | public  | `boolean`                      | `false`     | Sets disabled state for label                                     |
+| `size`     | public  | `"small"` `"medium"` `"large"` | `"medium"`  | Specifies font size for label                                     |
+| `weight`   | public  | `"regular"` `"semibold"`       | `"regular"` | Specifies font weight for label                                   |
+
+<br />
+
+### **Methods**
+
+<br />
+
+### **Events**
+
+<br />
+
+### **Attributes**
+
+| Name       | Field    |
+| ---------- | -------- |
+| `for`      | for      |
+| `from`     | from     |
+| `required` | required |
+| `disabled` | disabled |
+| `size`     | size     |
+| `weight`   | weight   |
+| `id`       | id       |
+
+<br />
+
+### **Slots**
+
+| Name | Description                            |
+| ---- | -------------------------------------- |
+|      | Default slotted content for label text |
+
+<br />
+<hr />
+<br />
+
+### **Suggested Template**
+
+```ts
+`<template
+    disabled="${(attr: Label) => attr.disabled}"
+    required="${(attr: Label) => attr.required}"
+    size="${(attr: Label) => attr.size}"
+    weight="${(attr: Label) => attr.weight"}>
+    <label
+      class="label"
+      part="label"
+      id="${(attr: Label) => attr.id}"
+      for="${(attr: Label) => attr.for}"
+      form="${(attr: Label) => attr.form}">
```

Comment:
This isn't going to work because we're shadowing an element which necessitates access to the same DOM as the element it is referencing. Given that this is shadowed it exists in its own DOM. I don't think we can bake in the label element here.